cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/MacroParser.java

```diff
@@ -146,17 +163,21 @@ private int getEndIndex(String str, int startIndex) {
   private int getArgsStartIndex(String str, String macroStr) {
     int argsStartIndex = macroStr.indexOf('(');
     while (escaped(argsStartIndex, str)) {
```

Comment:
actually, isn't the logic a little buggy? `\\\(` should evaluate to `\(`, but escaped() and doublyEscaped() will both return true right?

I think you need to change the logic of isEscaped() to count the number of preceding '\', and return true if odd, false if even.